src/game_api/entity.cpp

```diff
@@ -1,898 +1,812 @@
+            }
+            for (auto& [id, post] : _hook_info.post_render)
+            {
+                post(entity);
+            }
+        },
+        0x3);
+}
+void Entity::set_pre_render(std::uint32_t reserved_callback_id, std::function<bool(Entity* self)> pre_render)
+{
+    EntityHooksInfo& hook_info = get_hooks();
+    if (hook_info.pre_render.empty() && hook_info.post_render.empty())
+    {
+        hook_render_callback(this, rendering_info);
+    }
+    hook_info.pre_render.push_back({reserved_callback_id, std::move(pre_render)});
+}
+void Entity::set_post_render(std::uint32_t reserved_callback_id, std::function<void(Entity* self)> post_render)
+{
+    EntityHooksInfo& hook_info = get_hooks();
+    if (hook_info.pre_render.empty() && hook_info.post_render.empty())
+    {
+        hook_render_callback(this, rendering_info);
+    }
+    hook_info.post_render.push_back({reserved_callback_id, std::move(post_render)});
+}
+void Entity::set_enable_turning(bool enabled)
+{
+    set_entity_turning(this, enabled);
+}
+
+void Entity::set_climbable(bool climbable)
+{
+    set_entity_climbable(this, climbable);
+}
+
+std::span<uint32_t> Entity::get_items()
+{
+    if (items.size)
+        return std::span<uint32_t>(items.uids().begin(), items.uids().end());
+
+    return {};
+}
+
+Entity* get_entity_ptr(uint32_t uid)
+{
+    auto state = State::get();
+    auto p = state.find(uid);
+    if (IsBadWritePtr(p, 0x178))
+        return nullptr;
+    return p;
+}
+
+Entity* get_entity_ptr_local(uint32_t uid)
+{
+    auto state = State::get();
+    auto p = state.find_local(uid);
+    if (IsBadWritePtr(p, 0x178))
+        return nullptr;
+    return p;
+}
+
+std::vector<uint32_t> Movable::get_all_behaviors()
+{
+    std::vector<uint32_t> anims;
+    anims.reserve(behaviors_map.size());
+
+    for (auto& cur : behaviors_map)
+    {
+        anims.push_back(cur.first);
+    }
+    return anims;
+}
+
+bool Movable::set_behavior(uint32_t an)
+{
+    const auto& it = behaviors_map.find(an);
+    if (it != behaviors_map.end())
+    {
+        if (current_behavior != nullptr)
+        {
+            current_behavior->on_exit(this);
+        }
+        current_behavior = it->second;
+        if (current_behavior != nullptr)
+        {
+            current_behavior->on_enter(this);
+        }
+        return true;
+    }
+    return false;
+}
+
+uint32_t Movable::get_behavior()
+{
+    for (auto& cur : behaviors_map)
+    {
+        if (cur.second == current_behavior)
+        {
+            return cur.first;
+        }
+    }
+    return 0; // there is no id 0, but i can be wrong
+}
+
+void Movable::set_gravity(float gravity)
+{
+    hook_vtable<void(Movable*, float)>(
+        this,
+        [gravity](Movable* ent, [[maybe_unused]] float _gravity, void (*original)(Movable*, float))
+        {
+            original(ent, gravity);
+        },
+        0x53);
+}
+
+void Movable::reset_gravity()
+{
+    unregister_hook_function((void***)this, 0x53);
+}
```

Comment:
What's changed in this file? github doesn't show it well

Reply:
Only the facing left and climbable functions.

Reply:
And moving out stuff to the other files (mentioned in the description)